contrib/mesos/pkg/executor/service/service.go

```diff
@@ -288,8 +381,20 @@ func (ks *KubeletExecutorServer) createAndInitKubelet(
 		MaxContainers:      kc.MaxContainerCount,
 	}
 
+	// create main pod source
 	pc := kconfig.NewPodConfig(kconfig.PodConfigNotificationIncremental, kc.Recorder)
 	updates := pc.Channel(MESOS_CFG_SOURCE)
+	go func() {
+		for u := range execUpdates {
```

Comment:
can we get rid of KubeletExecutor.done? add a `defer close(someDoneChan)` here since `execUpdates` closes upon termination?